contracts/ScheduledPaymentModule.sol

```diff
@@ -78,6 +97,54 @@ contract ScheduledPaymentModule is Module {
         emit ScheduledPaymentCancelled(spHash);
     }
 
+    function executeScheduledPayment(
+        address token,
+        uint256 amount,
+        address payee,
+        uint256 maxGasPrice,
+        address gasToken,
+        uint256 _nonce,
+        uint256 payAt,
+        uint256 gasPrice
+    ) external onlyCrank {
+        bytes32 spHash = createSpHash(
+            token,
+            amount,
+            payee,
+            maxGasPrice,
+            gasToken,
+            _nonce,
+            payAt
+        );
+        if (!spHashes.contains(spHash)) revert UnknownHash(spHash);
+        if (block.timestamp < payAt.add(1 minutes))
```

Comment:
Shouldn't this be `block.timestamp >= payAt`? Meaning the scheduled payment gets executed only when `payAt` passes. 

I think it would also be useful to add a comment on why we're adding the time buffer. AFAIK it's to protect against miners gaming block time but I'm unsure what exactly that means

Reply:
This uses `if` not `require`, I need the code to revert the InvalidPeriod error when `block.timestamp < payAt.add(buffer)`. So I think it's already correct.

I will add the comment to explain about it.

Reply:
Oh yeah sorry, I see the logic now, looks ok.

Reply:
hmmm, so `block.timestamp` is a rather dangerous call to use. it's a classic front running vulnerability as well as miners can manipulate this. if we need to be reliant on a timestamp we should consider using an oracle to get the timestamp. either that or using block.number to measure time.

Reply:
I chose one minute as a buffer because the Ethereum mainnet POW consensus finality is waiting for the next six blocks, which means about 1 minute. I chose mainnet finality because I believe Ethereum POW consensus is the slowest of the other EVM chain consensus. But I'm not sure if 1 minute is too late or not for our scheduling.